engine/src/build/config/apple/apple_sdk.gni

```diff
@@ -2,16 +2,168 @@
+    ios_simulator_sdk_path = ""
+
+    # The path to iOS SDK Swift libraries.
+    ios_swift_lib_paths = []
+
+    # Version of iOS that we're targeting for tests.
+    ios_testing_deployment_target = "13.0"
+  }
+
+  # Must be set above or by gn.
+  assert(defined(use_ios_simulator))
+  assert(defined(ios_use_simulator))
+  assert(ios_deployment_target != "")
+  assert(ios_testing_deployment_target != "")
+}
+
+# Run apple_sdk.gni to determine SDK paths if necessary.
+_need_apple_sdk_run = false
+if (mac_host_toolchain_path == "" ||  #
+    mac_sdk_path == "" ||  #
+    mac_platform_path == "") {
+  _need_apple_sdk_run = true
+}
+if (is_ios) {
+  if (use_ios_simulator &&
+      (ios_simulator_platform_path == "" || ios_simulator_sdk_path == "")) {
+    _need_apple_sdk_run = true
+  }
+  if (!use_ios_simulator &&
+      (ios_device_platform_path == "" || ios_device_sdk_path == "")) {
+    _need_apple_sdk_run = true
+  }
+}
+if (_need_apple_sdk_run) {
+  _args = [ "--print-paths" ]
+  if (use_rbe && create_xcode_symlinks) {
+    # RBE has a restriction that paths cannot come from outside the build root.
+    _args += [
+      "--symlink",
+      rebase_path("//flutter/prebuilts"),
+    ]
+  }
+  _sdk_result =
+      exec_script(rebase_path("//build/mac/apple_sdk.py"), _args, "scope")
 }
 
-if (apple_host_toolchain_path == "") {
-  # Outputs the SDK path on the first line, and the host toolchain path on the
-  # second.
-  _find_sdk_lines =
-      exec_script("//build/mac/find_sdk.py", ["--print_sdk_path", "1.0"], "list lines")
+# Set SDK paths.
+if (mac_host_toolchain_path == "") {
+  mac_host_toolchain_path = _sdk_result.toolchain_path
+}
+assert(mac_host_toolchain_path != "")
+
+if (mac_sdk_path == "") {
+  mac_sdk_path = _sdk_result.macosx_sdk_path
+}
+assert(mac_sdk_path != "")
+
+if (mac_platform_path == "") {
+  mac_platform_path = _sdk_result.macosx_platform_path
+}
+assert(mac_platform_path != "")
+
+if (is_ios) {
+  if (ios_platform_path == "" || ios_sdk_path == "") {
+    if (use_ios_simulator) {
+      if (ios_simulator_platform_path == "") {
+        ios_simulator_platform_path = _sdk_result.iphonesimulator_platform_path
+      }
+      if (ios_simulator_sdk_path == "") {
+        ios_simulator_sdk_path = _sdk_result.iphonesimulator_sdk_path
+      }
+    } else {
+      if (ios_device_platform_path == "") {
+        ios_device_platform_path = _sdk_result.iphoneos_platform_path
+      }
+      if (ios_device_sdk_path == "") {
+        ios_device_sdk_path = _sdk_result.iphoneos_sdk_path
+      }
+    }
+
+    if (use_ios_simulator) {
+      assert(ios_simulator_platform_path != "")
+      assert(ios_simulator_sdk_path != "")
+      ios_platform_path = ios_simulator_platform_path
+      ios_sdk_path = ios_simulator_sdk_path
+    } else {
+      assert(ios_device_platform_path != "")
+      assert(ios_device_sdk_path != "")
+      ios_platform_path = ios_device_platform_path
+      ios_sdk_path = ios_device_sdk_path
```

Comment:
am i reading it right that the main purpose of the script is to set these 2 variables (`ios_platform_path` and `ios_sdk_path`), and they will be used in other scripts?

Reply:
Correct. These resolve to either the device or simulator platform/sdk depending on the value of `use_ios_simulator`.

Then, these two variables (and others) are used in BUILD.gn files as needed. e.g. you can see here, where I use the platform path in a followup patch:
https://github.com/flutter/flutter/pull/167958/files#diff-b39833088c1c25e24a99de123c9947c52eb9277d6fa0ec470ebc54444f13ff6fR42